docs/rules/no-undef.md

```diff
@@ -54,6 +58,23 @@ if (typeof UndefinedIdentifier === "undefined") {
 }
 ```
 
+#### typeof
+
+You can use this option if you want to prevent `typeof` check on a variable which has not been declared.
+
+The following patterns are considered problems with option `typeof` set:
+
+```js
+if(typeof a === "string"){}
```

Comment:
Can you show the actual configuration:

`/*eslint no-undef: [2, { typeof: true }]*/`